### PR TITLE
Fix key_value_field translation (#640)

### PR DESCRIPTION
**Pull request by yorch**
# Description

https://github.com/avo-hq/avo/pull/640

Fixes # (issue)

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the [documentation](https://docs.avohq.io)
- [ ] I have added tests that prove my fix is effective or that my feature works

## Manual review steps

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

1. Step 1
1. Step 2

Manual reviewer: please leave a comment with output from the test if that's the case.

### CODE REVIEW

**Comment by adrianthedev** [comment]
@yorch I'd like to fix this another way. Maybe with proper localization tags instead of passing them to the field.

File: lib/avo/fields/key_value_field.rb [hunk 1]
```diff
@@ -32,6 +28,30 @@ def initialize(id, **args, &block)
         @disable_deleting_rows = args[:disable_deleting_rows].present? ? args[:disable_deleting_rows] : false
       end
 
+      def key_label
+        return @key_label if @key_label && !@key_label.empty?
```
Comment (adrianthedev) [inline lib/avo/fields/key_value_field.rb:32]: Can we use `.present?` here and below?

```suggestion
        return @key_label if @key_label.present?
```

File: lib/avo/fields/key_value_field.rb [hunk 2]
```diff
@@ -15,10 +11,10 @@ def initialize(id, **args, &block)
 
         hide_on :index
 
-        @key_label = args[:key_label].present? ? args[:key_label].to_s : I18n.translate('avo.key_value_field.key')
-        @value_label = args[:value_label].present? ? args[:value_label].to_s : I18n.translate('avo.key_value_field.value')
-        @action_text = args[:action_text].present? ? args[:action_text].to_s : I18n.translate('avo.key_value_field.add_row')
-        @delete_text = args[:delete_text].present? ? args[:delete_text].to_s : I18n.translate('avo.key_value_field.delete_row')
+        @key_label = args[:key_label].to_s if args[:key_label].present?
```
Comment (adrianthedev) [inline lib/avo/fields/key_value_field.rb:14]: Do we **need** to cast to string in this case?

```suggestion
        @key_label = args[:key_label] if args[:key_label].present?
```

**Review by adrianthedev — APPROVED** [review]
I just have a few nitpicks regarding the coding style. We can merge after these are addressed.

File: lib/avo/fields/key_value_field.rb [hunk 3]
```diff
@@ -15,10 +11,10 @@ def initialize(id, **args, &block)
...
+        @value_label = args[:value_label].to_s if args[:value_label].present?
```
Comment (adrianthedev) [inline lib/avo/fields/key_value_field.rb:15]:
```suggestion
        @value_label = args[:value_label]if args[:value_label].present?
```

File: lib/avo/fields/key_value_field.rb [hunk 4]
```diff
@@ -15,10 +11,10 @@ def initialize(id, **args, &block)
...
+        @action_text = args[:action_text].to_s if args[:action_text].present?
```
Comment (adrianthedev) [inline lib/avo/fields/key_value_field.rb:16]:
```suggestion
        @action_text = args[:action_text] if args[:action_text].present?
```

File: lib/avo/fields/key_value_field.rb [hunk 5]
```diff
@@ -15,10 +11,10 @@ def initialize(id, **args, &block)
...
+        @delete_text = args[:delete_text].to_s if args[:delete_text].present?
```
Comment (adrianthedev) [inline lib/avo/fields/key_value_field.rb:17]:
```suggestion
        @delete_text = args[:delete_text] if args[:delete_text].present?
```

File: lib/avo/fields/key_value_field.rb [hunk 6]
```diff
@@ -32,6 +28,30 @@ def initialize(id, **args, &block)
...
       end
 
+      def key_label
+        return @key_label if @key_label && !@key_label.empty?
+
+        I18n.translate('avo.key_value_field.key')
+      end
+
+      def value_label
+        return @value_label if @value_label && !@value_label.empty?
```
Comment (adrianthedev) [inline lib/avo/fields/key_value_field.rb:38]:
```suggestion
        return @value_label if @value_label.present?
```

File: lib/avo/fields/key_value_field.rb [hunk 7]
```diff
@@ -32,6 +28,30 @@ def initialize(id, **args, &block)
...
+
+        I18n.translate('avo.key_value_field.value')
+      end
+
+      def action_text
+        return @action_text if @action_text && !@action_text.empty?
```
Comment (adrianthedev) [inline lib/avo/fields/key_value_field.rb:44]:
```suggestion
        return @action_text if @action_text.present?
```

File: lib/avo/fields/key_value_field.rb [hunk 8]
```diff
@@ -32,6 +28,30 @@ def initialize(id, **args, &block)
...
+
+        I18n.translate('avo.key_value_field.add_row')
+      end
+
+      def delete_text
+        return @delete_text if @delete_text && !@delete_text.empty?
```
Comment (adrianthedev) [inline lib/avo/fields/key_value_field.rb:50]:
```suggestion
        return @delete_text if @delete_text.present?
```

**Comment by yorch** [comment]
Sorry, just saw your messages, thanks for fixing the PR and merge it!

**Pull request merged**